### PR TITLE
Fix: android sdk accept-licenses returns non-zero when user declines

### PR DESCRIPTION
**Pull request by engyebrahim**
sdkmanager exits 0 even when the user types 'n'. After running the interactive acceptance, verify via AreLicensesAcceptedAsync (on-disk license file) and return exit code 1 if licenses were not accepted.
fixing: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2996356/

### CODE REVIEW

**Comment by engyebrahim** [comment]
The review is complete. Here's the summary:

## PR #319 Review: ✅ Ship it

**The fix is correct.** The `sdkmanager --licenses` tool exits 0 even when the user declines — so the PR switches to verifying license acceptance by re-checking the on-disk file via `AreLicensesAcceptedAsync()` instead of trusting the exit code.

### Non-blocking observations:
1. **Misleading test comment** — says "simulates user typing 'n'" but actually hits the non-interactive CI path
2. **Unused return value** — `RunInteractiveLicenseAcceptanceAsync` still returns `Task<int>` though both callers discard it; consider `Task` or `_ = await`
3. Tests pass, edge cases covered, consistent across both `accept-licenses` and `install` flows

File: src/Cli/Microsoft.Maui.Cli/Commands/AndroidCommands.Sdk.cs [hunk 1]
```diff
@@ -454,15 +454,18 @@ void WriteGroupedPackages(IEnumerable<SdkPackage> pkgs, string title)
 					formatter.WriteInfo("Starting interactive license acceptance...");
 					formatter.WriteInfo("Review each license and type 'y' to accept.\n");
 
-					var exitCode = await RunInteractiveLicenseAcceptanceAsync(androidProvider, cancellationToken);
-					if (exitCode == 0)
+					await RunInteractiveLicenseAcceptanceAsync(androidProvider, cancellationToken);
```
[the remark this answers is not in the document]
Comment (rmarinho) [reply src/Cli/Microsoft.Maui.Cli/Commands/AndroidCommands.Sdk.cs]: @copilot  can you apply this fix, log this, but keep the new logic?

**Comment by engyebrahim** [comment]
@rmarinho The interface doesn't currently allow us to log the failure diagnostic without the user seeing it, which would result in users seeing an exit code 0 even in non-acceptance cases. However, in vscode-maui, we only check if the exit code is 0, so I think it's fine to leave it as is. What are your thoughts?

**Review by rmarinho — APPROVED** [review]
